software/32.Zig/README.md

```diff
@@ -0,0 +1,267 @@
+✔️ How to install Zig
+✔️ Create your first project
+✔️ Discover the language's key concepts by creating a few projects
+
+## Introduction
+
+Zig is a general-purpose programming language focused on robustness, performance, and simplicity. It offers manual memory management, safety features, built-in cross-compilation, and seamless C interoperability, making it ideal for system programming and high-performance applications.
+
+## Step 0 - SETUP
+
+All the required information to install dependencies can be found in [SETUP.md](./SETUP.md).
+
+> 💡 We recommend you to follow the [Getting started](https://ziglang.org/learn/getting-started/) for this workshop.
+
+## Step 1 - Hello World! in Zig
+
+> ❗ We strongly advise you to use the documentation provided for this exercise.
+
+📑 Description:
+
+For the first exercise, we simply ask you to write `Hello world!` in your terminal when you run your program.
+
+📌 Tasks:
+
+Create a file `main.zig` in a folder called `src` with your logic to print the "hello world"
+
+src
+└── main.zig
+
+📚 Documentation:
+
+> 💡 Now, that you have created a file `main.zig`, you can use other zig files. To use them in your `main.zig` you have to integrate the module ([read more](https://stackoverflow.com/questions/71186556/how-do-i-include-one-zig-file-from-another-zig-file))
+
+- [Build System](https://ziglang.org/learn/build-system/)
+- [Doc.Zig](https://ziglang.org/documentation/master/)
+
+✔️ Validation:
+
+    you should see the following :
+
+    ```sh
+    Hello, World!
+    ```
+
+## Step 2 - Palindrome?
+
+📑 Description:
+
+For the second exercise, you have to create a function that takes as parameter a string `word`.
+
+📌 Tasks:
+
+Create a file `palindrome.zig` for this new function.
+
+This function must return true if the word given in parameter is a palindrome and false in the opposite case.
+
+📚 Documentation:
+
+ -[What is a palindrome ?](https://www.wikiwand.com/en/Palindrome)
+ -[New function](https://ziglang.org/documentation/master/#Functions)
+ -[The types in Zig](https://ziglang.org/documentation/master/#Primitive-Types)
+ -[Control_Structures](https://zig.guide/language-basics/while-loops)
+
+ ✔️ Validation:
+
+    Here's a main example :
+
+    pub fn main() void {
+    const stdout = std.io.getStdOut().writer();
+
+    const test_word1 = "madam";
+    const test_word2 = "hello";
+
+    const result1 = is_palindrome(test_word1);
+    const result2 = is_palindrome(test_word2);
+
+    stdout.print("{} is palindrome: {}\n", .{test_word1, result1}) catch {};
+    stdout.print("{} is palindrome: {}\n", .{test_word2, result2}) catch {};
+}
+
+ When you compile and run palindrome.zig, the output should be:
+
+    ```sh
+    madam is palindrome: true
+    ```
+    or
+
+    ```sh
+    hello is palindrome: false
+    ```
+
+## Step 3 - Fibonacci sequence
```

Comment:
As I already said guys, change this exercise and find a new one, it's already been done in the rust workshop.